docs/course_materials/exercises/test_exercises/04_Compound_data_types/test_homework_lists.py

```diff
@@ -0,0 +1,109 @@
+         '1.1. Expected the list to be set by using `= []`'],
+        [r'''^list.*\s*=\s*list\(\)''',
+         '1.1. Expected the list to be set by using list()'],
+        # Create a list l1 with the numbers: 2,3,4,5,6.
+        [r'''^l1\s*=\s*\[2,\s*3,\s*4,\s*5,\s*6\]''',
+         '1.2. list l1 is not defined properly'],
+        # Print the index of the number 6
+        [r'''l1\.index\(6\)''',
+         '1.3. Expected the use of list.index()'],
+        # Append the numbers 7 and 8 to the end of l1
+        [r'''l1\.(extend|append)\(\[?7.*''',
+         '1.4. Expected list.extend([]) or list.append() to add values'],
+        # Print the length of the l1
+        [r'''len\(l1\)''',
+         '1.5. Expected len(list) for getting the length of the list'],
+        # Add the number 1 to the l1 at index 0
+        [r'''l1\.insert\(0, 1\)''',
+         '1.6. Expected the use of list.insert(index, value)'],
+        # comment for Slicing a list exercise
+        [r'''^[#]+.*((S|s)licing|2)''',
+         '2. Expected a comment for exercise 2 - Slicing a list'],
+        # Print first 4 numbers of l1
+        [r'''(print\()?l1\[0?:4\]\)?''',
+         '2.1. Expected the use of range for printing a sequence'],
+        # Print 4 last numbers of l1
+        [r'''(print\()?l1\[-4:\]\)?''',
+         '2.2. Expected the use of negative range for printing a sequence'],
+        # Print the two numbers in the middle
+        [r'''(print\()?l1\[((3:5)|(\(len\(l1\)\/2\-1\):''' +
+         r'''\(len\(l1\)\/2\+1\)))\]\)''',
+         '2.3. Expected the use of len(list) in retrieving the middle values'],
+        # Print l1[20], which error did you get?
+        # What does this error say? (write your answer in a comment #)
+        [r'''^[#].*IndexError: list index out of range''',
+         '2.4. Expected the `IndexError: ...`'],
+        # comment for Update a list exercise
+        [r'''^[#]+.*((U|u)pdate|3)''',
+         '3. Expected a comment for exercise 3 - Update the list'],
+        # Remove the items 4 and 5 from the l1 and print l1
+        [r'''(del l1\[3:5\])|(l1.remove\(4\))''',
+         '3.1 Expected the use of `del list[:]` or list.remove(i)'],
+        # Create a new list l2, with the items: -1, -2 ,-3 and print l2
+        [r'''^l2\s*=\s*\[\-1,\s*\-2,\s*\-3\]''',
+         '3.2. list l2 is not defined properly'],
+        # Create a new list l3, with the items of l1 and l2, print l3
+        [r'''(l3\s*\=\s*l1\s*\+\s*l2)|(l3\.extend\(l1\))''',
+         '3.3. Expected l3 to be defined by l1 and l2'],
+        # Sort l3 and print the sorted list
+        [r'''(print\()?l3\.sort\(\)''',
+         '3.4. Expected the use of list.sort() for sorting a list'],
+        # Print the length of l1, l2, and l3
+        [r'''(print\()?.*\blen\(l1\).*\)?''',
+         '3.5. Expected the use of len(list)'],
+    ]
+
+    helpers.expected_stdout = [
+        [r'\b4\b[^\]]$',
+         '1.3. Index of number 6 is incorrect'],
```

Comment:
a student work: 
l1=[2,3,4,5,6]
print(l1.index(6))
ans she still got: 
1.3. Index of number 6 is incorrect